library/std/src/f128/soft/tests.rs

```diff
@@ -0,0 +1,49 @@
+use crate::f128::soft::{Positive, PositiveFinite, Representation, Sign, pow2};
+
+#[test]
+fn test_representation() {
+    assert_eq!(Representation::new(-1.5f128), Representation {
+        sign: Sign::Negative,
+        abs: Positive::Finite(PositiveFinite { exp: -112, mantissa: 3 << 111 }),
+    });
+
+    assert_eq!(Representation::new(f128::MIN_POSITIVE), Representation {
+        sign: Sign::Positive,
+        abs: Positive::Finite(PositiveFinite { exp: -16494, mantissa: 1 << 112 }),
+    });
+
+    assert_eq!(Representation::new(f128::MIN_POSITIVE / 2.0), Representation {
+        sign: Sign::Positive,
+        abs: Positive::Finite(PositiveFinite { exp: -16495, mantissa: 1 << 112 }),
```

Comment:
This `1 << 112` is a meaningful magic number, and should be derivable from the existing constants (`SIGNIFICAND_BITS` and so on).

Can we lift this into a `const MANTISSA_TOP_BIT` (or whatever) that computes it? If you want to also `assert_eq!` that const against `1 << 112` still, that's fine too.